models/quickbooks__general_ledger_by_period.sql

```diff
@@ -19,6 +31,21 @@ final as (
 
     select *
     from retained_earnings
+),
+
+final as (
+
+    select balances_earnings_unioned.*,
+        coalesce(account_number_ordinal.ordinal, account_sub_type_ordinal.ordinal, account_type_ordinal.ordinal, account_class_ordinal.ordinal) as account_ordinal
+    from balances_earnings_unioned
+        left join ordinals as account_class_ordinal
+            on balances_earnings_unioned.account_class = account_class_ordinal.account_class
+        left join ordinals as account_type_ordinal
+            on balances_earnings_unioned.account_type = account_type_ordinal.account_type
+        left join ordinals as account_sub_type_ordinal
+            on balances_earnings_unioned.account_sub_type = account_sub_type_ordinal.account_sub_type
+        left join ordinals as account_number_ordinal
+            on balances_earnings_unioned.account_number = account_number_ordinal.account_number
```

Comment:
Similar to my above comment in this model, I would recommend we rework this to only leverage the joins if the ordinal variable is configured. We can then take advantage of our defaults in a single `case when` statement if the variable is not defined. My thought is it would look something like the following:

```sql
    select 
        balances_earnings_unioned.*,
    {% if var('financial_statement_ordinal') %}
        coalesce(account_number_ordinal.ordinal, account_sub_type_ordinal.ordinal, account_type_ordinal.ordinal, account_class_ordinal.ordinal) as account_ordinal
    {% else %}
        case 
            when account_class = "Asset" then 1
            when account_class = "Liability" then 2
            when account_class = "Equity" then 3
            when account_class = "Revenue" then 1
            when account_class = "Expense" then 2
        end as account_ordinal 
    {% endif %}
    from balances_earnings_unioned
    {% if var('financial_statement_ordinal') %}
        left join ordinals as account_class_ordinal
            on balances_earnings_unioned.account_class = account_class_ordinal.account_class
        left join ordinals as account_type_ordinal
            on balances_earnings_unioned.account_type = account_type_ordinal.account_type
        left join ordinals as account_sub_type_ordinal
            on balances_earnings_unioned.account_sub_type = account_sub_type_ordinal.account_sub_type
        left join ordinals as account_number_ordinal
            on balances_earnings_unioned.account_number = account_number_ordinal.account_number
    {% endif %}
```

This way we can bake in a redundancy if the variable is not defined and used by the customer. The best of both worlds with allowing the customization to be used only if the customer wants to.

Reply:
Updated!